src/pages/Models/Details/ModelDetails.js

```diff
@@ -124,48 +125,62 @@ const ModelDetails = () => {
   }, [dispatch]);
 
   useEffect(() => {
-    if (modelUUID !== null && filteredModelStatusData === null) {
+    if (modelUUID !== null && modelStatusData === null) {
       // This model may not be in the first batch of models that we request
       // status from in the main loop so update the status now.
       dispatch(fetchModelStatus(modelUUID));
     }
-  }, [dispatch, modelUUID, filteredModelStatusData]);
+  }, [dispatch, modelUUID, modelStatusData]);
 
-  const handleRowClick = (e) => {
-    const currentTarget = e.currentTarget;
-    if (currentTarget.classList.contains("is-selected")) {
-      setFilterByApp("");
-      return;
-    }
-    setFilterByApp(currentTarget.dataset.app);
+  const handleAppRowClick = (e, app) => {
+    const currentApp = cloneDeep(app);
+    currentApp.name = e.currentTarget.dataset.app;
+    setSlidePanelData({ currentApp });
+    setFilterByApp(currentApp.name);
   };
 
   const viewFilters = ["all", "apps", "units", "machines", "relations"];
 
   const applicationTableRows = useMemo(
     () =>
-      generateApplicationRows(
-        filteredModelStatusData,
-        filterByApp,
-        handleRowClick,
-        baseAppURL
-      ),
-    [baseAppURL, filterByApp, filteredModelStatusData]
+      generateApplicationRows(modelStatusData, handleAppRowClick, baseAppURL),
+    [baseAppURL, modelStatusData]
   );
+
   const unitTableRows = useMemo(
-    () => generateUnitRows(filteredModelStatusData, filterByApp, baseAppURL),
-    [baseAppURL, filterByApp, filteredModelStatusData]
+    () => generateUnitRows(modelStatusData, baseAppURL),
+    [baseAppURL, modelStatusData]
+  );
+  const unitSlidePanelRows = useMemo(
```

Comment:
It might be nice to add a small comment to these lines that are going to be removed (or moved) when this is refactored as in the comment above.